arch/arm/core/aarch64/smccc-call.S

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2019 Carlo Caione <ccaione@baylibre.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <toolchain.h>
+#include <linker/sections.h>
+#include <arch/cpu.h>
+
```

Comment:
Just a note here would help someone unfamiliar to the fact that it implements a spec. 
The SMCCC  excerpt to show what parameter means what would also help :) .  Being greedy here.

Reply:
you are totally right in being greedy here ;) I'll add some comments.

Reply:
@sandeepbrcm I added some comments and a link to the spec. Let me know if you want me to add anything else.